server/pkg/wfexec/session.go

```diff
@@ -689,7 +688,18 @@ func (s *Session) exec(ctx context.Context, log *zap.Logger, st *State) (nxt []*
 				zap.Error(st.err),
 			)
 
-			_ = expr.Assign(scope, "error", expr.Must(expr.NewString(st.err.Error())))
+			ehr := st.results
+			if ehr != nil {
+				if _, ok := ehr.GetValue()["error"]; ok {
+					_ = expr.Assign(scope, ehr.GetValue()["error"].Get().(string), expr.Must(expr.NewAny(st.err)))
+				}
+				if _, ok := ehr.GetValue()["errorMessage"]; ok {
+					_ = expr.Assign(scope, ehr.GetValue()["errorMessage"].Get().(string), expr.Must(expr.NewString(st.err.Error())))
+				}
+				if _, ok := ehr.GetValue()["errorStepID"]; ok {
+					_ = expr.Assign(scope, ehr.GetValue()["errorStepID"].Get().(string), expr.Must(expr.NewInteger(st.step.ID())))
+				}
```

Comment:
Let's reduce the number of `GetValue` calls.
The change probably won't make much of a difference in performance, but still.